### PR TITLE
Tired Employee Enemy

### PR DESCRIPTION
**Pull request by jamesdrayton**
This is the end result of the Trello task "Tired Employee Enemy" (https://trello.com/c/q986X7BT)

It takes a duplicate of the player and replaces the PlayerController script with the EmployeeController script. The tired employee enemy has modifiable moveSpeed and damage by default, as well as an array of empty waypoint objects which it will iterate through to create a path. It will always begin at the first waypoint in the array before moving on to the next, and if Looping is set to true then it will also return to the first waypoint upon reaching the last. If false, it will stop moving at the last waypoint. 

Besides leaving room to apply a sprite and animations to the employee, there is also room for the employee to do something after reaching each waypoint, and to apply different effects upon colliding with other objects.

### CODE REVIEW

File: EGaDS! Open Project Game/Assets/Scripts/EmployeeController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    [SerializeField] private Transform[] waypoints;
+    // controls the movespeed of the employee
+    [SerializeField] private float moveSpeed = 10;
+    // true if the employee loops waypoints, false if it stops at the last waypoint 
+    [SerializeField] private bool looping = true;
+    // controls the amount of damage dealt to the player upon touching the employee
+    [SerializeField] private int damage = 1;
+    private int nextWaypoint = 1;
+    public Rigidbody2D rb;
```
Comment (jamesouyang2003) [inline EGaDS! Open Project Game/Assets/Scripts/EmployeeController.cs:15]: Is `rb` ever used? If not, then this line should probably be removed.

File: EGaDS! Open Project Game/Assets/Scripts/EmployeeController.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    // Update is called once per frame
+    void Update() {
+        checkWaypoint();
+        transform.position = Vector2.MoveTowards(transform.position, waypoints[nextWaypoint].position, moveSpeed * Time.deltaTime);
+    }
+
+    // Checks if Employee has reached nextWaypoint and updates nextWaypoint
+    void checkWaypoint() {
```
Comment (jamesouyang2003) [inline EGaDS! Open Project Game/Assets/Scripts/EmployeeController.cs:30]: The naming convention for methods are `MethodName()`, so this should be `CheckWaypoint()`

**Review by jamesouyang2003 — CHANGES_REQUESTED** [review]
Thanks for working on the enemy! It all looks great for a basic enemy AI. I've added some comments in your EmployeeController script, which are mostly just some really simple changes you can make. 

Another thing I see is that the enemy might rotate sometimes when colliding with something, so in the RigidBody2D, go to "Constraints", then check the z axis for "Freeze Rotation". 

Finally, it would also be great if you could make a prefab for the tired employee in the Assets/Prefabs folder by dragging the game object into that folder in Unity. The lets us add more copies of the enemy by dragging from the folder to the scene, so it'd be a great help!

**Comment by jamesdrayton** [comment]
All done on the changes

**Pull request merged**